src/main/scala/is/hail/variant/GATKGenotypeView.scala

```diff
@@ -0,0 +1,265 @@
+package is.hail.variant
+
+import is.hail.annotations.{MemoryBuffer, RegionValue, UnsafeRow, UnsafeUtils}
+import is.hail.expr._
+import is.hail.utils._
+
+object GATKGenotypeView {
+  def apply(rowSignature: TStruct): GATKGenotypeView = {
+    rowSignature.fields(2).typ.asInstanceOf[TArray].elementType match {
+      case TGenotype => new TGenotypeView(rowSignature)
+      case _: TStruct => new StructGenotypeView(rowSignature)
+      case t => fatal(s"invalid genotype representation: $t, expect TGenotype or TStruct")
+    }
+  }
+}
+
+// FIXME: This is removed, and StructGenotypeView becomes the only genotype view when TGenotype is removed
+sealed abstract class GATKGenotypeView {
```

Comment:
I think this should be called NGSGenotypeView.  Genomes on the Cloud also generates roughly this this schema (has OD but not DP?  I forget exactly.)